packages/nx-plugin/src/plugin/plugin.unit.test.ts

```diff
@@ -0,0 +1,66 @@
+import { CreateNodesContext } from '@nx/devkit';
+import { vol } from 'memfs';
+import { describe, expect } from 'vitest';
+import { createFilesAndInvokeCreateNodesOnThem } from '@code-pushup/test-utils';
+import { PROJECT_JSON_FILE_NAME } from '../internal/constants';
+import { CP_TARGET_NAME } from './constants';
+import { createNodes } from './plugin';
+
+describe('@code-pushup/nx-plugin/plugin', () => {
+  let context: CreateNodesContext;
+
+  beforeEach(() => {
+    context = {
+      nxJsonConfiguration: {},
+      workspaceRoot: '',
+    };
+  });
+
+  afterEach(() => {
+    vol.reset();
+  });
+
+  it('should normalize context and use it to create target on ROOT project', async () => {
+    const projectRoot = '.';
+    const matchingFilesData = {
+      [`${projectRoot}/${PROJECT_JSON_FILE_NAME}`]: `${JSON.stringify({
+        name: '@org/empty-root',
+      })}`,
+    };
+    const projects = await createFilesAndInvokeCreateNodesOnThem(
+      createNodes,
+      context,
+      {},
+      { matchingFilesData },
+    );
+
+    // project should have one target created
+    const targets = projects[projectRoot]?.targets ?? {};
```

Comment:
There's a type error because the test util incorrectly claims to resolve to `CreateNodesResult`:

![image](https://github.com/user-attachments/assets/8db964b7-ab06-4fa5-a9d5-2c283ba8d31c)

Reply:
I adjusted the types. Have a look.

Reply:
No, that's still not accurate. The code un-nests `projects`, but doesn't match the `CreatesNodeResult` type. 

If you want to resolve to `CreatesNodeResult`, then the merged projects should be under `projects`:

![image](https://github.com/user-attachments/assets/c4428bcf-7b84-42be-b642-d243f9d101af)

In this case, usage would be `const { projects } = await invokeCreateNodesOnVirtualFiles(...)`.

Also, not sure if you want to ignore the `externalNodes` this way instead of merging them from the results (same as for `projects`).

Reply:
Sry for the comment before. I deleted it and wanted to come up with a better solution. 
Thanks for the information I get back to it later.

Reply:
fixed!! thx for the second detailed answer! Should have understood it earlier :)